homeassistant/components/watergate/sensor.py

```diff
@@ -0,0 +1,274 @@
+"""Support for Watergate sensors."""
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from datetime import UTC, datetime
+import logging
+
+from watergate_local_api.models import AutoShutOffReport
+
+from homeassistant.components.sensor import (
+    HomeAssistant,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import (
+    SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+    EntityCategory,
+    UnitOfPressure,
+    UnitOfTemperature,
+    UnitOfTime,
+    UnitOfVolume,
+    UnitOfVolumeFlowRate,
+)
+from homeassistant.core import callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import WatergateConfigEntry
+from .coordinator import WatergateAgregatedRequests, WatergateDataCoordinator
+from .entity import WatergateEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 0
+
+ASO_EVENT_TYPE_ATRIBUTE = "type"
+ASO_EVENT_VOLUME_ATRIBUTE = "volume"
+ASO_EVENT_DURATION_ATRIBUTE = "duration"
+
+UPTIME_ENTITY_NAME = "uptime"
+POWER_SUPPLY_ENTITY_NAME = "power_supply"
+PRESSURE_ENTITY_NAME = "water_pressure"
+TEMPERATURE_ENTITY_NAME = "water_temperature"
+FLOW_ENTITY_NAME = "water_flow_rate"
+IP_ENTITY_NAME = "ip"
+GATEWAY_ENTITY_NAME = "gateway"
+SUBNET_ENTITY_NAME = "subnet"
+SSID_ENTITY_NAME = "ssid"
+RSSI_ENTITY_NAME = "rssi"
+WIFI_UPTIME_ENTITY_NAME = "wifi_uptime"
+MQTT_UPTIME_ENTITY_NAME = "mqtt_uptime"
+SHUT_OFF_EVENT_ENTITY_NAME = "auto_shut_off_event"
+WATER_METER_VOLUME_ENTITY_NAME = "water_meter_volume"
+WATER_METER_DURATION_ENTITY_NAME = "water_meter_duration"
+
+
+@dataclass(kw_only=True, frozen=True)
+class WatergateSensorEntityDescription(SensorEntityDescription):
+    """Description for Acaia sensor entities."""
+
+    value_fn: Callable[[WatergateAgregatedRequests], str | int | float | None]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: WatergateConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up all entries for Watergate Platform."""
+
+    coordinator = config_entry.runtime_data
+
+    descriptions: list[WatergateSensorEntityDescription] = [
+        WatergateSensorEntityDescription(
+            value_fn=lambda data: data.state.water_meter.duration
+            if data.state and data.state.water_meter
+            else None,
+            translation_key=WATER_METER_VOLUME_ENTITY_NAME,
+            key=WATER_METER_VOLUME_ENTITY_NAME,
+            native_unit_of_measurement=UnitOfVolume.LITERS,
+            device_class=SensorDeviceClass.WATER,
+            state_class=SensorStateClass.TOTAL_INCREASING,
+        ),
```

Comment:
Please move the entity descriptions OUT of async_setup, into a constant.
Please also drop all single-use constants to have the key and translation_key as regular strings.